options_test.go

```diff
@@ -25,13 +25,123 @@ import (
 	models "github.com/firecracker-microvm/firecracker-go-sdk/client/models"
 )
 
+func TestGetFirecrackerConfig(t *testing.T) {
+	tempFile, err := ioutil.TempFile("", "firectl-test-drive-path")
+	if err != nil {
+		t.Error(err)
+	}
+	defer func() {
+		tempFile.Close()
+		os.Remove(tempFile.Name())
+	}()
+	cases := []struct {
+		name        string
+		opts        *options
+		expectedErr func(error) (bool, error)
+		outConfig   firecracker.Config
+	}{
+		{
+			name: "Invalid metadata",
+			opts: &options{
+				FcMetadata: "{ invalid:json",
+			},
+			expectedErr: func(e error) (bool, error) {
+				return strings.HasPrefix(e.Error(), invalidMetadata.Error()), invalidMetadata
+			},
+			outConfig: firecracker.Config{},
+		},
+		{
+			name: "Invalid network config",
+			opts: &options{
+				FcNicConfig: "no-slash",
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == parseNicConfigError, parseNicConfigError
+			},
+			outConfig: firecracker.Config{},
+		},
+		{
+			name: "Invalid drives",
+			opts: &options{
+				FcNicConfig:        "a/b",
+				FcAdditionalDrives: []string{"/no-suffix"},
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == invalidDriveSpecificationNoSuffix, invalidDriveSpecificationNoSuffix
+			},
+			outConfig: firecracker.Config{},
+		},
+		{
+			name: "Invalid vsock addr",
+			opts: &options{
+				FcNicConfig:        "a/b",
+				FcAdditionalDrives: []string{tempFile.Name() + ":ro"},
+				FcVsockDevices:     []string{"noCID"},
+			},
+			expectedErr: func(e error) (bool, error) {
+				return e == unableToParseVsockDevices, unableToParseVsockDevices
+			},
+			outConfig: firecracker.Config{},
+		},
+		{
+			name: "Invalid fifo config",
+			opts: &options{
+				FcNicConfig:        "a/b",
+				FcAdditionalDrives: []string{tempFile.Name() + ":ro"},
+				FcVsockDevices:     []string{"a:3"},
+				FcFifoLogFile:      "/etc/we-dont-have-access",
```

Comment:
What if we do have access?

Reply:
The wrong thing. I modified this so that access/lack of access isn't part of the test.